src/Microsoft.Azure.WebJobs.Host/Bindings/OpenType.cs

```diff
@@ -6,28 +6,328 @@
 
 namespace Microsoft.Azure.WebJobs.Host.Bindings
 {
+    // Ensures that open Types match consistently 
+    public class OpenTypeMatchContext
+    {
+        internal Type _match;
+
+        internal bool CheckArg(Type type)
```

Comment:
This is a public type, with no public members?

Reply:
This is making sure that when using OpenTypes for both src and dest, they match consistently.  For example, suppose we register a converter:
     IEnumerable<T> --> Widget<T> 
We need to make sure that T is the same in both src and dest. So we have a "context" that is shared across both OpenTypes matches and it verifies consistency. 
The methods themselves are non-public since derived OpenTypes don't need to worry about the exact contract, just that the instance of the contract is the same. 

---
In reply to: [150930248](https://github.com/Azure/azure-webjobs-sdk/pull/1422#discussion_r150930248) [](ancestors = 150930248)